salesforce/events.go

```diff
@@ -0,0 +1,341 @@
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := common.UnmarshalJSON[SFAPIResponseBody](resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.Warnings != nil && len(res.Warnings) > 0 {
+		slog.Warn("CreateEventChannelMember", "warnings", res.Warnings)
+	}
+
+	channel.Id = res.Id
+
+	return channel, nil
+}
+
+func (c *Connector) CreateEventChannelMember(
+	ctx context.Context,
+	member *EventChannelMember,
+) (*EventChannelMember, error) {
+	location, err := joinURLPath(c.BaseURL, "tooling/sobjects/PlatformEventChannelMember")
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.post(ctx, location, member)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := common.UnmarshalJSON[SFAPIResponseBody](resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.Warnings != nil && len(res.Warnings) > 0 {
+		slog.Warn("CreateEventChannelMember", "warnings", res.Warnings)
+	}
+
+	member.Id = res.Id
+
+	return member, nil
+}
+
+// nolint: lll
+// https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_eventrelayconfig.htm?q=EventRelayConfig
+func (c *Connector) CreateEventRelayConfig(
+	ctx context.Context,
+	cfg *EventRelayConfig,
+) (*EventRelayConfig, error) {
+	location, err := joinURLPath(c.BaseURL, "tooling/sobjects/EventRelayConfig")
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.post(ctx, location, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := common.UnmarshalJSON[SFAPIResponseBody](resp)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Id = res.Id
+
+	return cfg, nil
+}
+
+// nolint: lll
+// https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_eventrelayconfig.htm?q=EventRelayConfig
+func (c *Connector) RunEventRelay(ctx context.Context, configId string) (*EventRelayConfig, error) {
+	location, err := joinURLPath(c.BaseURL, "tooling/sobjects/EventRelayConfig", configId)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &EventRelayConfig{
+		Id: configId,
+		Metadata: &EventRelayConfigMetadata{
+			State: "RUN",
+		},
+	}
+
+	resp, err := c.patch(ctx, location, config)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := common.UnmarshalJSON[SFAPIResponseBody](resp)
+	if err != nil {
+		return nil, err
+	}
+
+	config.Id = res.Id
+
+	return config, nil
+}
+
+func (c *Connector) getOrganization(ctx context.Context) (map[string]*ajson.Node, error) {
+	location, err := joinURLPath(c.BaseURL, "connect/organization")
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.get(ctx, location)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Body.GetObject()
+}
+
+func (c *Connector) GetOrganizationId(ctx context.Context) (string, error) {
+	org, err := c.getOrganization(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return org["orgId"].MustString(), nil
+}
+
+func (c *Connector) GetRemoteResouece(ctx context.Context, channelId string) (string, error) {
+	orgId, err := c.GetOrganizationId(ctx)
```

Comment:
These methods should not keep calling `GetOrganizationId` otherwise we'll be unnecessarily calling the same API many times. This should take in orgId as a parameter. Same for the other methods in this file that are calling `GetOrganizationId`